openapi-cli/src/main/java/io/ballerina/generators/GeneratorUtils.java

```diff
@@ -251,18 +252,20 @@ public static String convertOpenAPITypeToBallerina(String type) {
                 convertedType = "[]";
                 break;
             case Constants.OBJECT:
-                convertedType = "record";
+                convertedType = "record {}";
                 break;
             case Constants.DECIMAL:
                 convertedType = "decimal";
                 break;
             case Constants.NUMBER:
+                convertedType = "decimal";
+                break;
             case Constants.DOUBLE:
             case Constants.FLOAT:
                 convertedType = "float";
                 break;
             default:
-                convertedType = "anydata";
+                throw new BallerinaOpenApiException("Encountered an unsupported swagger data type `" + type + "`");
```

Comment:
`Encountered an unsupported swagger data type` should change to `Unsupported OAS data type`